File: Graphs/find_vertex_edge.py
# File Name: find_vertex_edge.py

graph = []
vertices = int(input("Enter number of vertices: "))

for i in range(vertices):
    graph.append([])

def find_vertex(v):
    if 0 <= v < vertices:
        print(f"Vertex {v} exists in the graph.")
    else:
        print("Vertex does not exist!")

def find_edge(u, v):
    if u >= vertices or v >= vertices:
        print("Invalid vertices!")
        return

    if v in graph[u]:
        print(f"Edge exists between {u} and {v}.")
    else:
        print("Edge does not exist!")

def display_graph():
    print("\nAdjacency List:")
    for i in range(vertices):
        print(f"{i} -> {graph[i]}")

while True:
    print("\n1. Find Vertex")
    print("2. Find Edge")
    print("3. Display Graph")
    print("4. Exit")
    choice = int(input("Enter your choice: "))

    if choice == 1:
        v = int(input("Enter vertex to find: "))
        find_vertex(v)
    elif choice == 2:
        u, v = map(int, input("Enter two vertices (u v): ").split())
        find_edge(u, v)
    elif choice == 3:
        display_graph()
    elif choice == 4:
        break
    else:
        print("Invalid choice")

File: Graphs/find_vertex_edge.c
/* File Name: find_vertex_edge.c */

#include <stdio.h>
#include <stdlib.h>

#define MAX 100

int graph[MAX][MAX];
int vertices;

void findVertex(int v) {
    if (v >= 0 && v < vertices)
        printf("Vertex %d exists in the graph.\n", v);
    else
        printf("Vertex does not exist!\n");
}

void findEdge(int u, int v) {
    if (u >= vertices || v >= vertices) {
        printf("Invalid vertices!\n");
        return;
    }

    if (graph[u][v] == 1)
        printf("Edge exists between %d and %d.\n", u, v);
    else
        printf("Edge does not exist!\n");
}

void displayGraph() {
    printf("\nAdjacency Matrix:\n");
    for (int i = 0; i < vertices; i++) {
        for (int j = 0; j < vertices; j++)
            printf("%d ", graph[i][j]);
        printf("\n");
    }
}

int main() {
    int choice, u, v;

    printf("Enter number of vertices: ");
    scanf("%d", &vertices);

    while (1) {
        printf("\n1. Find Vertex\n2. Find Edge\n3. Display Graph\n4. Exit\n");
        printf("Enter your choice: ");
        scanf("%d", &choice);

        switch (choice) {
            case 1:
                printf("Enter vertex to find: ");
                scanf("%d", &v);
                findVertex(v);
                break;
            case 2:
                printf("Enter two vertices (u v): ");
                scanf("%d %d", &u, &v);
                findEdge(u, v);
                break;
            case 3:
                displayGraph();
                break;
            case 4:
                exit(0);
            default:
                printf("Invalid choice\n");
        }
    }
}
